src/corehost/cli/deps_resolver.cpp

```diff
@@ -555,30 +559,38 @@ void deps_resolver_t::resolve_additional_deps(const hostpolicy_init_t& init)
         }
         else
         {
-            // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
-            append_path(&additional_deps_path, _X("shared"));
-            append_path(&additional_deps_path, fx_name.c_str());
-            append_path(&additional_deps_path, fx_ver.c_str());
-
-            // The resulting list will be empty if 'additional_deps_path' is not a valid directory path
-            std::vector<pal::string_t> list;
-            pal::readdir(additional_deps_path, _X("*.deps.json"), &list);
-            for (pal::string_t json_file : list)
+            for (int i = 1; i < m_fx_definitions.size(); ++i)
             {
-                pal::string_t json_full_path = additional_deps_path;
-                append_path(&json_full_path, json_file.c_str());
-                m_additional_deps_files.push_back(json_full_path);
-
-                trace::verbose(_X("Using specified additional deps.json: '%s'"), 
-                    json_full_path.c_str());
+                // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
+                pal::string_t additional_deps_path_fx = additional_deps_path;
+                append_path(&additional_deps_path_fx, _X("shared"));
+                append_path(&additional_deps_path_fx, m_fx_definitions[i]->get_name().c_str());
+                append_path(&additional_deps_path_fx, m_fx_definitions[i]->get_found_version().c_str());
```

Comment:
//review note: in the original code this was the version from the config (fx_ver), not the found version (which seems incorrect in the original code)